src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -46,6 +46,7 @@ PreReleaseRequest<I>::~PreReleaseRequest() {
   if (!m_shutting_down) {
     m_image_ctx.state->handle_prepare_lock_complete();
   }
+  delete m_on_finish;
```

Comment:
>It is expected behaviour for Context derived classes that they delete themselves on complete

@trociny It doesn't check that way. It checks only: "the destructor and whatever functions it calls do not free it."